### PR TITLE
refactor: Split out func GetSettings from FindRequirements

### PR DESCRIPTION
**Pull request by msvticket**
To ease reuse I'm splitting out GetSettings from FindRequirements. I will use GetSettings to get the issueTracker in jx-changelog.

### CODE REVIEW

File: pkg/variablefinders/requirements.go [hunk 1]
```diff
@@ -105,6 +65,54 @@ func FindRequirements(g gitclient.Interface, jxClient jxc.Interface, ns, dir, ow
 	return req, nil
 }
 
+func GetSettings(g gitclient.Interface, jxClient jxc.Interface, ns, dir, owner, repo string) (*jxcore.Settings, string, error) {
```
Comment (ankitm123) [inline pkg/variablefinders/requirements.go:68]: can you add a godoc comment about what the function does, similar to what `FindRequirements` has.

File: pkg/variablefinders/requirements.go [hunk 2]
```diff
@@ -13,61 +13,21 @@ import (
...
-	if err != nil {
-		return nil, errors.Wrapf(err, "failed to load source configs")
-	}
-	groupSettings := sourceconfigs.FindSettings(srcConfig, owner, repo)
-
-	settings, err = mergeSettings(settings, groupSettings)
+	settings, clusterDir, err := GetSettings(g, jxClient, ns, dir, owner, repo)
 	if err != nil {
-		return nil, errors.Wrapf(err, "failed to merge settings")
+		return nil, err
```
Comment (ankitm123) [inline pkg/variablefinders/requirements.go:21]: can we do error.Wrap(err, "something like failed to get settings from ....")?
Might help some users to get more context around errors.

File: pkg/variablefinders/requirements.go (hunk 1 above)
Comment (msvticket) [reply pkg/variablefinders/requirements.go:69]: OK

File: pkg/variablefinders/requirements.go (hunk 2 above)
Comment (msvticket) [reply pkg/variablefinders/requirements.go]: OK

**Pull request merged**